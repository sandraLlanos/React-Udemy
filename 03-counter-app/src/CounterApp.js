import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value = 10 }) => {
    
    // const state = useState('Goku');
    // console.log( state ); // ["Goku", f]

    // const [ nombre, setNombre ] = useState('Goku');
    // console.log(nombre, setNombre);

    const [counter, setCounter] = useState( value );
    const handleAdd = () => {
        setCounter( counter + 1 ); // I'm modifying state hook not counter constant
        // setCounter( (count) => count + 1 ) // this is other way
    }
    // const handleAdd = (e) => {
    //     console.log(e);
    // }

    const handleSubtract = () => {
        setCounter(counter - 1 );
    }
    const handleReset = () => {
        setCounter(value);
    }
    return (
        <div className="wrapper">
           <h1>CounterApp</h1>
           <h1> Initial Value:{ value } </h1> 
           <h2> { counter } </h2> 
            <div>
                {/* <button onClick={ (e) => handleAdd(e) }>+1</button> */}
                <button onClick={ handleSubtract }>-1</button>
                <button onClick={ handleReset }>Reset</button>
                <button onClick={ handleAdd }>+1</button>
            </div>
        </div>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number,
}

export default CounterApp
