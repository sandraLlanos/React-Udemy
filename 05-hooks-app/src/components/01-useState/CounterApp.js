import React, { useState } from 'react';
import './counter.css';

export const CounterApp = () => {
    const [state, setState] = useState({
        counter1:10,
        counter2:20,
        counter3:30,
        counter4:40
    });
    const {counter1, counter2} = state;
    const handledCounter = () => {
        setState({
            ...state,
            counter1: counter1 + 1
        })
    }
    return (
        <>
            <h1>Counter { counter1 } </h1>   
            <h1>Counter { counter2 } </h1>   
            <hr/>
            <button className="btn btn-primary" 
                    onClick={handledCounter}>
                +1
            </button>
        </>
    )
}
