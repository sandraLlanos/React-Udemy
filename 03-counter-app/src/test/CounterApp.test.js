import { shallow } from "enzyme"
import CounterApp from "../CounterApp"

describe('test in <CounterApp /> component', () => {
    
    let wrapper = shallow( <CounterApp /> );
    beforeEach( () => {
        wrapper = shallow( <CounterApp /> );
        console.log('beforeEach');
    });

    test('should show <CounterApp /> correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    test('should show <CounterApp /> with value for default', () => {
        const val = 100;
        const wrapper = shallow( <CounterApp value={val}/> );
        const counterText = wrapper.find('h2').text().trim();
        // console.log(counterText);
        expect( counterText ).toBe(`${val}`);
    });
    test('should increase with a button +1', () => {
        // const btnOne = wrapper.find('button').at(2);
        // console.log(btnOne.html());
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect( counterText ).toBe('11');
    });
    test('should decrease with a button -1', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect( counterText ).toBe('9');
    });
    test('should reset value with a button reset', () => {
        const val = 100;
        const wrapper = shallow( <CounterApp value={val}/> );
        wrapper.find('button').at(2).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);
        expect( counterText ).toBe(`${val}`);
    });
    
    
})
