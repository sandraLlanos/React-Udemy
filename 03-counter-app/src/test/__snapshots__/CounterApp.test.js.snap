// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <CounterApp /> component should show <CounterApp /> correctly 1`] = `
<div
  className="wrapper"
>
  <h1>
    CounterApp
  </h1>
  <h1>
     Initial Value:
    10
     
  </h1>
  <h2>
     
    10
     
  </h2>
  <div>
    <button
      onClick={[Function]}
    >
      -1
    </button>
    <button
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      onClick={[Function]}
    >
      +1
    </button>
  </div>
</div>
`;
